src/configuration.cpp

```diff
@@ -161,152 +163,172 @@ chimera::CompiledConfiguration::CompiledConfiguration(
+        if (bindingNode)
+        {
+            // Check that 'binding' node in configuration YAML is a scalar.
+            if (!typesNode.IsScalar())
             {
-                std::cerr << "Unable to resolve type: "
-                          << "'" << type_str << "'" << std::endl;
+                std::cerr << "'binding' in configuration YAML must be a map."
+                          << std::endl;
                 exit(-2);
             }
+            binding_name = bindingNode.as<std::string>();
         }
     }
 
-    // Construct the binding that is used for this configuration.
-    // If a configuration is not found, the default values will be taken from the preset binding.
-    // TODO: We can now very easily handle multiple default bindings in the
-    //       future by changing the next line to depend on the configuration :)
-    templateBinding_ = PYTHON_BINDING;
-    if (templateNode_)
+    // Resolve the base binding definition from the specified binding name.
+    const auto bindingIt = chimera::binding::DEFINITIONS.find(binding_name);
+    if (bindingIt == chimera::binding::DEFINITIONS.end())
+    {
+        std::cerr << "Unable to resolve binding definition: "
+                  << "'" << binding_name << "'" << std::endl;
+        exit(-2);
+    }
+    bindingDefinition_ = bindingIt->second;
+
+    // Override individual templates if specified in the configuration.
+    if (bindingNode_)
     {
-        if (const YAML::Node &classTemplateNode = templateNode_["class"])
-            templateBinding_.class_cpp = Lookup(classTemplateNode);
+        if (const YAML::Node &classTemplateNode = bindingNode_["class"])
+            bindingDefinition_.class_cpp = Lookup(classTemplateNode);
 
-        if (const YAML::Node &enumTemplateNode = templateNode_["enum"])
-            templateBinding_.enum_cpp = Lookup(enumTemplateNode);
+        if (const YAML::Node &enumTemplateNode = bindingNode_["enum"])
+            bindingDefinition_.enum_cpp = Lookup(enumTemplateNode);
 
-        if (const YAML::Node &functionTemplateNode = templateNode_["function"])
-            templateBinding_.function_cpp = Lookup(functionTemplateNode);
+        if (const YAML::Node &functionTemplateNode = bindingNode_["function"])
+            bindingDefinition_.function_cpp = Lookup(functionTemplateNode);
 
-        if (const YAML::Node &moduleTemplateNode = templateNode_["module"])
-            templateBinding_.module_cpp = Lookup(moduleTemplateNode);
+        if (const YAML::Node &moduleTemplateNode = bindingNode_["module"])
+            bindingDefinition_.module_cpp = Lookup(moduleTemplateNode);
 
-        if (const YAML::Node &variableTemplateNode = templateNode_["variable"])
-            templateBinding_.variable_cpp = Lookup(variableTemplateNode);
+        if (const YAML::Node &variableTemplateNode = bindingNode_["variable"])
+            bindingDefinition_.variable_cpp = Lookup(variableTemplateNode);
```

Comment:
I'm pretty sure all of these `YAML::Node`s should be stored by value, not `const &`.